algebra/tests/quat/various.c

```diff
@@ -1053,3 +1066,84 @@ TEST_GROUP_RUNNER(group_quat_vecRot)
 	RUN_TEST_CASE(group_quat_vecRot, quat_vecRot_biggerValues);
 	RUN_TEST_CASE(group_quat_vecRot, quat_vecRot_zeroVector);
 }
+
+
+/* ##############################################################################
+ * ------------------        quat_rotQuat tests       --------------------
+ * ############################################################################## */
+
+
+TEST_GROUP(group_quat_rotQuat);
+
+
+TEST_SETUP(group_quat_rotQuat)
+{
+}
+
+
+TEST_TEAR_DOWN(group_quat_rotQuat)
+{
+}
+
+
+TEST(group_quat_rotQuat, quat_rotQuat_baseQuaternions)
+{
+	quat_t q, qA, qI, qJ, qK, qZero = { 0 };
+	const vec_t versX = { .x = 1.0f, .y = 0.0f, .z = 0.0f };
+	const vec_t versY = { .x = 0.0f, .y = 1.0f, .z = 0.0f };
+	const vec_t versZ = { .x = 0.0f, .y = 0.0f, .z = 1.0f };
+
+	qA = qI = qJ = qK = qZero;
+	qA.a = qI.i = qJ.j = qK.k = 1.0f;
+
+	/* Rotation about 0 angle - quaternion qA */
+	quat_rotQuat(&versX, 0.0, &q);
+
+	TEST_ASSERT_QUAT_WITHIN(DELTA, qA, q);
+
+	/* Rotation about 180 degrees along x-axis - quaternion qI */
+	quat_rotQuat(&versX, M_PI, &q);
+
+	TEST_ASSERT_QUAT_WITHIN(DELTA, qI, q);
+
+	/* Rotation about 180 degrees along y-axis - quaternion qJ */
+	quat_rotQuat(&versY, M_PI, &q);
+
+	TEST_ASSERT_QUAT_WITHIN(DELTA, qJ, q);
+
+	/* Rotation about 180 degrees along z-axis - quaternion qK */
+	quat_rotQuat(&versZ, M_PI, &q);
+
+	TEST_ASSERT_QUAT_WITHIN(DELTA, qK, q);
```

Comment:
Little convention breaking, but much easier to read:
```suggestion
	/* Rotation about 0 angle - quaternion qA */
	quat_rotQuat(&versX, 0.0, &q);
	TEST_ASSERT_QUAT_WITHIN(DELTA, qA, q);

	/* Rotation about 180 degrees along x-axis - quaternion qI */
	quat_rotQuat(&versX, M_PI, &q);
	TEST_ASSERT_QUAT_WITHIN(DELTA, qI, q);

	/* Rotation about 180 degrees along y-axis - quaternion qJ */
	quat_rotQuat(&versY, M_PI, &q);
	TEST_ASSERT_QUAT_WITHIN(DELTA, qJ, q);

	/* Rotation about 180 degrees along z-axis - quaternion qK */
	quat_rotQuat(&versZ, M_PI, &q);
	TEST_ASSERT_QUAT_WITHIN(DELTA, qK, q);
```